src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,104 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private StorageDaoImpl storage;
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    public void setUp() {
+        storage = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_nullUser_notOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("password");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("user");
+        user.setPassword(null);
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin("user");
+        user.setPassword("password");
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
```

Comment:
The test case `register_nullAge_notOk` is problematic because the `age` field in the `User` class is likely an `int`, which cannot be null. This test will not compile if `age` is an `int`. Consider removing this test or adjusting it to test for invalid age values that are possible with the data type used.